edx_lint/pylint/filters_docstring/filters_docstring_check.py

```diff
@@ -0,0 +1,117 @@
+    DOCSTRING_MISSING_TYPE = "filter-docstring-missing-type"
+    DOCSTRING_MISSING_TRIGGER = "filter-docstring-missing-trigger"
+
+    msgs = {
+        ("E%d91" % BASE_ID): (
+            "Filter's (%s) docstring is missing the required purpose section or is badly formatted",
+            DOCSTRING_MISSING_PURPOSE,
+            "filters docstring is missing the required purpose section or is badly formatted",
+        ),
+        ("E%d92" % BASE_ID): (
+            "Filter's (%s) docstring is missing the required type section or is badly formatted",
+            DOCSTRING_MISSING_TYPE,
+            "filters docstring is missing the required type section or is badly formatted",
+        ),
+        ("E%d93" % BASE_ID): (
+            "Filter's (%s) docstring is missing the required trigger section or is badly formatted",
+            DOCSTRING_MISSING_TRIGGER,
+            "filters docstring is missing the required trigger section or is badly formatted",
+        ),
+    }
+
+    options = ()
+
+    @utils.only_required_for_messages(
+        DOCSTRING_MISSING_PURPOSE,
+        DOCSTRING_MISSING_TYPE,
+        DOCSTRING_MISSING_TRIGGER,
+    )
+    def visit_classdef(self, node):
+        """
+        Visit a class definition and check its docstring.
+
+        If the class is a subclass of OpenEdxPublicFilter, check the format of its docstring. Skip the
+        OpenEdxPublicFilter class itself.
+
+        """
+        if not node.is_subtype_of("openedx_filters.tooling.OpenEdxPublicFilter"):
+            return
+
+        if node.name == "OpenEdxPublicFilter":
+            return
+
+        docstring = node.doc_node.value if node.doc_node else ""
+        if not (error_messages := self._check_docstring_format(docstring)):
+            return
+        for error_message in error_messages:
+            self.add_message(error_message, node=node, args=(node.name,))
+
+    def _check_docstring_format(self, docstring):
+        """
+        Check the format of the docstring for errors and return a list of error messages.
+
+        The docstring should have the following structure:
+        1. Description: Any non-empty text followed by a blank line.
+        2. Filter Type: A line that starts with "Filter Type:".
+        3. Trigger: A line that starts with "Trigger:".
+
+        For example:
+
+        ```
+        Description:
+        Filter used to modify the certificate rendering process.
+
+        ... (more description)
+
+        Filter Type:
+            org.openedx.learning.certificate.render.started.v1
+
+        Trigger:
+            - Repository: openedx/edx-platform
+            - Path: lms/djangoapps/certificates/views/webview.py
+            - Function or Method: render_html_view
+        ```
+        """
+        required_sections = [
+            (r"Purpose:\s*.*\n", self.DOCSTRING_MISSING_PURPOSE),
+            (r"Filter Type:\s*.*\n", self.DOCSTRING_MISSING_TYPE),
```

Comment:
We are currently just validating that it's not empty, shouldn't we make this regex more robust to ensure it matches a valid type? Or do we not have a defined set of valid types yet?

Reply:
Yes, we have a format for the filter_type attribute (see [ADR-0004](https://docs.openedx.org/projects/openedx-filters/en/latest/decisions/0004-filters-naming-and-versioning.html#decisions)), but even though, there are a lot of cases to consider. A good middle ground to still allow flexibility could be to check that the `Filter Type` from the docstring matches the `OpenEdxPublicFilter.filter_type`, so we know that docstrings contain accurate information. I can try applying this suggestion to openedx-events as well.